src/sync.rs

```diff
@@ -501,10 +501,7 @@ fn sync_submission_chain<R, S>(
                         ),
                         this_state.stages,
                     ));
-                } else if !prev.exclusive() {
-                    // Insert barrier here.
-                    // Prev won't insert as it isn't exclusive.
-                    assert!(this.exclusive());
+                } else {
```

Comment:
"Multiple queries executing the acquire side"? I thought every buffer in a `Link` would be run in the same queue, so this would work fine. Am I wrong here?

Reply:
Oh, I see how it works.